contributing.md

```diff
@@ -128,11 +187,22 @@ To check that the package is compatible with server side rendering, build the pr
 yarn check:ssr
 ```
 
-### How to make an entropy source
+### How to add an entropy source
 
-Entropy source is a function that gets a piece of data about the browser a.k.a. a fingerprint component.
+An entropy source is a function that gets a piece of data about the browser.
+The value returned by an entropy source (also called an entropy component) is used to produce the visitor identifier.
 Entropy sources are located in the [src/sources](src/sources) directory.
-Entropy component must be a simple JS value that can be JSON-encoded.
+All entropy components must be simple JavaScript values that can be encoded into JSON.
+
+Entropy source must meet the following requirements:
+- It is stable — it always or almost always produces the same value in each browser, including incognito, guest, and desktop modes.
+- It is selective — it produces different values in different browsers, operating systems, or devices.
+    A good entropy source represents the browser, operating system, or device settings.
+- It produces no side effects, such as messages in the browser console, DOM changes, modal windows, notifications, and sounds.
+- It is fast. An entropy source should take not more than 1 second to complete.
+- It doesn't represent only the browser version. An example of such a signal is a JavaScript feature probing.
+    It's not a good entropy source because the return value is likely to change due to automatic browser updates,
+    the selectivity is low, and the same entropy can be achieved by just using the User-Agent string.
 
 Entropy source runs in 2 stages: "load" and "get":
```

Comment:
```suggestion
The entropy source runs in 2 stages: "load" and "get":
```

Reply:
I suppose it's wrong because the text refers to entropy source as a concept, not a specific entropy source